extension/chrome/elements/attachment_preview.ts

```diff
@@ -79,9 +80,11 @@ View.run(
           });
           $('#attachment-preview-download')
             .css('display', 'flex')
-            .on('click', e => {
+            .on('click', async e => {
               e.stopPropagation();
-              Browser.saveToDownloads(attachmentForSave);
+              if (await AttachmentWarnings.confirmSaveToDownloadsIfNeeded(attachmentForSave)) {
```

Comment:
As `AttachmentPreview` takes the whole page, we don't need to use messaging here, a simple `Ui.modal.confirm` call will be done here